commit 6c8e110015c829e04d8f5fb2ac53b23f468d58ed
Author: Viktor Gal <viktor.gal@maeth.com>
Date:   Thu May 28 10:02:27 2009 +1000

    Fix for ticket 475: fix NULL pointer dereference in oggplay_seek_cleanup function.
    The fix proposed by Chris Double in annodex trac could cause another
    NULL pointer dereference, in case of not enough memory for allocating new buffer.

diff --git a/src/liboggplay/oggplay_seek.c b/src/liboggplay/oggplay_seek.c
index ef150b8..7d7073d 100644
--- a/src/liboggplay/oggplay_seek.c
+++ b/src/liboggplay/oggplay_seek.c
@@ -104,6 +104,9 @@ oggplay_seek_cleanup(OggPlay* me, ogg_int64_t milliseconds)
   /*
    * store the old buffer in it next.
    */
+  if (me->buffer == NULL)
+    return;
+  
   trash->old_buffer = (OggPlayBuffer *)me->buffer;
 
   /*
